src/seedu/addressbook/data/person/Address.java

```diff
@@ -22,11 +29,21 @@
      */
     public Address(String address, boolean isPrivate) throws IllegalValueException {
         String trimmedAddress = address.trim();
-        this.isPrivate = isPrivate;
-        if (!isValidAddress(trimmedAddress)) {
-            throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
+        
+        Pattern parserPattern = Pattern.compile(ADDRESS_VALIDATION_REGEX);
+        Matcher parserMatcher = parserPattern.matcher(trimmedAddress);
+
+        if (parserMatcher.find()) {
+           block = new Block(parserMatcher.group(1));
+           street = new Street(parserMatcher.group(3));
+           unit = new Unit(parserMatcher.group(5));
+           postalCode = new PostalCode(parserMatcher.group(7));
+        }
+        else {
+           throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = trimmedAddress;
+        
+        value = block.getValue() + ", " +  street.getValue() + ", " + unit.getValue() + ", " + postalCode.getValue();
```

Comment:
No need to save the input address in `value` as it may cause inconsistency due to duplication.